smtchecker.js

```diff
@@ -16,17 +16,37 @@ var potentialSolvers = [
 var solvers = potentialSolvers.filter(solver => commandExistsSync(solver.name));
 
 function solve (query) {
-  var tmpFile = tmp.fileSync();
+  var tmpFile = tmp.fileSync({ postfix: '.smt2' });
   fs.writeFileSync(tmpFile.name, query);
   // TODO For now only the first SMT solver found is used.
   // At some point a computation similar to the one done in
   // SMTPortfolio::check should be performed, where the results
   // given by different solvers are compared and an error is
   // reported if solvers disagree (i.e. SAT vs UNSAT).
-  var solverOutput = execSync(solvers[0].name + ' ' + solvers[0].params + ' ' + tmpFile.name);
+  var outputStr;
+  try {
+    var solverOutput = execSync(
+      solvers[0].name + ' ' + solvers[0].params + ' ' + tmpFile.name, {
+        timeout: 10000
+      }
+    );
+    outputStr = solverOutput.toString();
+  } catch (e) {
+    // execSync throws if the process times out or returns != 0.
+    // The latter might happen with z3 if the query asks for a model
+    // for an UNSAT formula. We can still use stdout.
+    outputStr = e.stdout.toString();
+    if (
+      !outputStr.startsWith('sat') &&
+      !outputStr.startsWith('unsat') &&
+      !outputStr.startsWith('unknown')
+    ) {
+      throw new Error('Failed solve SMT query. ' + e.toString());
```

Comment:
Is this a separate fix?

Reply:
This yes, the others are rather enhancements.

Reply:
Can you create a PR for these fixes? Perhaps on top of #325.

Reply:
https://github.com/ethereum/solc-js/pull/328